src/Xamarin.Android.Build.Tasks/Tasks/BuildApk.cs

```diff
@@ -371,11 +371,20 @@ public override bool RunTask ()
 			OutputFiles = outputFiles.Select (a => new TaskItem (a)).ToArray ();
 
 			Log.LogDebugTaskItems ("  [Output] OutputFiles :", OutputFiles);
-			DSOWrapperGenerator.CleanUp (BuildEngine4, Log);
+			DSOWrapperGenerator.CleanUp (this);
 
 			return !Log.HasLoggedErrors;
 		}
 
+		internal DSOWrapperGenerator.Config EnsureConfig ()
+		{
+			var config = BuildEngine4.GetRegisteredTaskObjectAssemblyLocal<DSOWrapperGenerator.Config> (ProjectSpecificTaskObjectKey (DSOWrapperGenerator.RegisteredConfigKey), RegisteredTaskObjectLifetime.Build);
```

Comment:
I didn't like moving this method here, but `ProjectSpecificTaskObjectKey()` is `protected`:
* https://github.com/dotnet/android-tools/blob/ab2165daf27d4fcb29e88bc022e0ab0be33aff69/src/Microsoft.Android.Build.BaseTasks/AndroidTask.cs#L34